src-srv/utils/transformations/yjs/yPlanning.ts

```diff
@@ -0,0 +1,69 @@
+import * as Y from 'yjs'
+import { type Document, type Block } from '../../../protos/service.js'
+
+export function newsDocToYPlanning(document: Document, planningYMap: Y.Map<unknown>): Y.Map<unknown> {
+  try {
+    const root = {
+      title: document.title
+    }
+    const meta = toYMap(groupBy(document.meta, 'type'), new Y.Map())
+    const links = toYMap(groupBy(document.links, 'type'), new Y.Map())
+
+    planningYMap.set('meta', meta)
+    planningYMap.set('links', links)
+    planningYMap.set('root', toYMap(root, new Y.Map()))
+    return planningYMap
+  } catch (err) {
+    if (err instanceof Error) {
+      throw new Error(err.message)
+    }
+    throw new Error('Unknown error')
+  }
+}
+
+type GroupedObjects = Record<string, Block[]>
+const groupBy = (objects: Block[], groupKey: keyof Block): GroupedObjects => {
+  const groupedObjects: GroupedObjects = {}
+
+  objects.forEach(object => {
+    const { [groupKey as keyof Block]: arrayToGroup, ...rest } = object
+    if (!arrayToGroup) return
+
+    if (!groupedObjects[(object[groupKey] as string)]) {
+      groupedObjects[(object[groupKey]) as string] = []
+    }
+
+    // eslint-disable-next-line
+    const newObj: any = { ...rest }
+
+    Object.keys(object).forEach((key) => {
+      if (Array.isArray(object[key as keyof Block])) {
+        const array = object[key as keyof Block]
+        newObj[key] = groupBy((array as Block[]), groupKey)
+      }
+    })
+
+    groupedObjects[(object[groupKey]) as string].push(newObj as Block)
+  })
+
+  return groupedObjects
+}
+
+// TODO: Almost identical to newsDocToYmap function, make reuseable?
```

Comment:
It seems at first glance as they are indeed almost the same. It would maybe be an idea to make this more generic and document it a little bit. Let's have a discussion around it.

Reply:
I'll look into that, tried to keep them separated at first so we wouldn't get any regressions in the handling of textbit editing. But I think we have some bugs (hopefully solved in this) in `newsDocToYmap`.